### ISSUE TITLE
app.auth is not a function

### ISSUE THREAD

**Issue by rtm** [angular/angularfire2#1707]
### Version info

**Angular:** 6.0.0

**Firebase:** 5.0.4

**AngularFire:** 5.0.0-rc.10

**Other (e.g. Ionic/Cordova, Node, browser, operating system):** WSL, Node 8.11.2, Cordova, etc.

I receive the message: 

> app.auth is not a function

This occurs in the following code on the line `return app.auth()`:

    var AngularFireAuth = (function () {
      function AngularFireAuth(options, nameOrConfig, platformId, zone) {
        var _this = this;
        this.zone = zone;
        var scheduler = new angularfire2__WEBPACK_IMPORTED_MODULE_3__["FirebaseZoneScheduler"](zone, platformId);
        this.auth = zone.runOutsideAngular(function () {
            var app = Object(angularfire2__WEBPACK_IMPORTED_MODULE_3__["_firebaseAppFactory"])(options, nameOrConfig);
            return app.auth();
        });

`app` seems to be well-formed:

![image](https://user-images.githubusercontent.com/24391/40901778-4e42c306-67ef-11e8-93af-f7e1fc1c1629.png)

but it has no `auth` property. I am instantiating (or rather, Angular is instantiating) `AngularFireAuth` because it is injected into a service, which itself is instantiated quite early in the application life cycle. Was there some change in how soon the `app` is completely ready, including its `auth` property? Or is this possibly some kind of version conflict?

**Comment by MartinNuc** [comment]
Finally I got rid of that thing. I am not exactly sure what was the cause but after upgrading to new angularfire2 I had some problems with imports and I googled to install some packages from `@firebase/...` after that I got `app.auth() is not a function.

Today I removed those dependencies. So related things in my `package.json` are:

```
"firebase": "5.0.4"
"angularfire2": "5.0.0-rc.10"
```

but then I got error: `Module not found: Error: Can't resolve '@firebase/webchannel-wrapper'`

So I installed only that `webchannel-wrapper`: `npm install @firebase/webchannel-wrapper --save-dev`

Remember to remove your `node_modules` and redoing `npm install` just to be sure.

In my code when I need something from `firebase` I use this kind of import: 

```
import { auth } from 'firebase/app';
import { FirebaseError } from 'firebase/app';
```

Hope it helps :-/

**Comment by mohamedshameer** [comment]
@MartinNuc what version of @firebase/app are you using

**Comment by MartinNuc** [comment]
firebase: 5.0.4
angularfire2: 5.0.0-rc.10

**Comment by hiepxanh** [comment]
version conflict was resolved with `firebase@5.1.0` and `angularfire2 rc 11`. Do you still have problem?

**Comment by mohamedshameer** [comment]
Thanks @hiepxanh its working fine now .

**Comment by hiepxanh** [comment]
@mohamedshameer moha good to know that!
@rtm do you still have that problem?

**Comment by ziyaddin** [comment]
@hiepxanh unfortunately, your version suggestion did not work. I personally use `angularfire2@5.0.0-rc.3` and `firebase@4.8.0` and it works fine. You can use it as an alternative.

**Comment by jburns24** [comment]
@hiepxanh I to have been running into this issue. using 
firebase: 5.2.0
angularfire2: 5.0.0-rc.11
angular: 6.0.7

I have a sample app that I was trying to make this work in that you can checkout. https://github.com/jburns24/angularfire2-auth-bad-demo

**Comment by Qarun-Qadir-Bissoondial** [comment]
@MartinNuc  When you imported auth from firebase/app, can you provide some code as to how you used the import?

**Comment by Kamshak** [comment]
For me in my bundle there were somehow two versions of @firebase/app: 
```
/node_modules/@firebase/app/dist/index.esm.js?89c8 
/node_modules/@firebase/app/dist/index.esm.js?cc84
```
to workaround this i added a plugin in my webpack config:
```
new NormalModuleReplacementPlugin(/\@firebase\/app(.*)/, require.resolve('@firebase/app').replace('index.node.cjs.js', 'index.esm.js'))
```

That fixed it for me. Versions:
```
"angularfire2": "5.0.0-rc.9",
"firebase": "5.3.0",
```

**Comment by Kamshak** [comment]
That doesn't work for prod though :cry:

**Comment by davideast** [comment]
@Kamshak What do you imports of Firebase look like?

**Comment by Kamshak** [comment]
@davideast 

Imports are done the normal way through AngularFire*Module imports and in one case I import auth like this to create auth providers:
```ts
import { auth, User as FirebaseUser } from 'firebase/app';
// some code
const provider = new auth.GoogleAuthProvider();
```

**Comment by davideast** [comment]
@Kamshak For TypeScript you should use the star import so nothing funky occurs.

```ts
import * as firebase from 'firebase/app';
```

Can you give that a try to and see if it helps? Firebase isn't packaged for single exports like you're using and that may be causing problems.

**Comment by Kamshak** [comment]
@davideast thanks for the reply! I upgraded webpack/angular alongside angularfire and it turns out the issue was that I had this in my webpack config:
``` ts
{
      test: /\.js$/,
      loader: 'source-map-loader',
      enforce: 'pre'
}
``` 
but you now need to exclude the ngfactory stuff:
```ts
{
      test: /\.js$/,
      loader: 'source-map-loader',
      enforce: 'pre',
      exclude: [ /(ngfactory|ngstyle).js$/ ]
}
``` 

Because otherwise for some reason different loaders would get applied to the module apparently which creates the 'query' part in the include. Sorry about spamming here a bit, was getting a bit frustrated with the issue and it's really difficult to get there from the error.

TL;DR: If you have this issue search in your main/vendor bundle for `@firebase/app/dist/index.esm.js` if you see something like this `@firebase/app/dist/index.esm.js?a42bfc` carefully check your webpack loaders.

**Comment by guillefd** [comment]
Had the same issue after upgrading to `angularfire2 5.0.0-rc-11` and `firebase 5.0.4`.  
Reinstalled node_modules, cleaned node cache, and `app.auth is not a function` keep showing.  

SOLUTION
After pretty much time debugging, did the following and it worked:
- created a new FRESH angular project
- installed angularfire and firebase `$ npm install firebase angularfire2 --save`
- replaced complete `/src` folder in new project with previous project `/src` folder.
- added required dependencies to package.json from previous project and run `npm install`
- run `$ ng serve` and worked.
